packages/firestore/lite/test/integration.test.ts

```diff
@@ -832,4 +836,134 @@ describe('Query', () => {
       }
     );
   });
+
+  it('supports collection groups', () => {
+    return withTestCollection(async collRef => {
+      const collectionGroupId = `${collRef.id}group`;
+
+      const fooDoc = doc(
+        collRef.firestore,
+        `${collRef.id}/foo/${collectionGroupId}/doc1`
+      );
+      const barDoc = doc(
+        collRef.firestore,
+        `${collRef.id}/bar/baz/boo/${collectionGroupId}/doc2`
+      );
+      await setDoc(fooDoc, { foo: 1 });
+      await setDoc(barDoc, { bar: 1 });
+
+      const query = collectionGroup(collRef.firestore, collectionGroupId);
+      const result = await getQuery(query);
+
+      verifyResults(result, { bar: 1 }, { foo: 1 });
+    });
+  });
+
+  it('validates collection groups', () => {
+    return withTestDb(firestore => {
+      expect(() => collectionGroup(firestore, '')).to.throw(
+        'Function collectionGroup() requires its first argument to be of type non-empty string, but it was: ""'
+      );
+      expect(() => collectionGroup(firestore, '/')).to.throw(
+        "Invalid collection ID '/' passed to function collectionGroup(). Collection IDs must not contain '/'."
+      );
+    });
+  });
+});
+
+describe('equality', () => {
+  it('for collection references', () => {
+    return withTestDb(firestore => {
+      const coll1a = collection(firestore, 'a');
+      const coll1b = parent(doc(firestore, 'a/b'));
+      const coll2 = collection(firestore, 'c');
+
+      expect(refEqual(coll1a, coll1b)).to.be.true;
```

Comment:
this is really 4 test cases that you are conflating into one it() block, consider splitting them out

Reply:
I understand, but these test cases are very related and overall still small. Having a very large number of small tests can also become a nuisance. Fortunately, no one on our team has TypeScript readability, so we can decide on a case by case basis. Some of this might go away as we add "context" sections.

FYI We will probably rewrite all these tests once we have a version of the full Firestore API that uses tree-shaking. We should then use the same tests against all codebases.